### PR TITLE
Add warn when input attributes is missing

### PR DESCRIPTION
**Pull request by aflmp**
Fixes: #4386

### CODE REVIEW

**Comment by aflmp** [comment]
Hello @anderseknert, does the approach look ok to you? If yes, I can work on adding/fixing tests.

**Review by anderseknert — CHANGES_REQUESTED** [review]
Thanks! Left a comment on the URL as I think we might want to skip that. Looks like several tests need to be updated to work with this addition though, and perhaps you'd want to add a test or two on your own, unless this is already well covered by those that are there already (I think that might be the case).

File: server/server.go [hunk 1]
```diff
@@ -1415,6 +1415,10 @@ func (s *Server) v1DataGet(w http.ResponseWriter, r *http.Request) {
 		DecisionID: decisionID,
 	}
 
+	if input == nil {
```
Comment (christian1607) [inline server/server.go:1418]: I think this validates if the 'input' http query parameter is present in the request, the validation should be after JSON  unmarshall is completed  and validate if it has an 'input' field.

Comment (anderseknert) [reply server/server.go:1418]: That's a good point, Christian! The tests should help shine some light on this :)

Comment (tsandall) [reply server/server.go:1418]: Making the server generate a warning on GET w/o an input document feels a bit wrong to me. It's not invalid to execute a policy query without supplying an input document. The POST endpoint is specifically for executing policies with an input document provided--so let's warn if users invoke that without supplying input--but for GET, I'll be a bit sad if running `curl localhost:8181/v1/data` complains at me 😿

**Review by tsandall — CHANGES_REQUESTED** [review]
Warning if the POST v1/data endpoint is called w/o providing `input` makes sense, but I'd rather not see that warning show up in GET v1/data.

**Comment by anderseknert** [comment]
Agreed @tsandall 👍 @aflmp let me know if you want to address the issues mentioned here or if someone should pick up from where we are.

**Comment by aflmp** [comment]
> Agreed @tsandall 👍 @aflmp let me know if you want to address the issues mentioned here or if someone should pick up from where we are.

thank you @tsandall @anderseknert, pushed the changes based on the feedback.

File: server/server_test.go [hunk 2]
```diff
@@ -1463,6 +1533,16 @@ p = true { false }`
 				]
 			  }`},
 		}},
+		{"post api usage warning", []tr{
```
Comment (aflmp) [inline server/server_test.go:1536]: added tests for warning but I can remove them if they look redundant

**Review by anderseknert — APPROVED** [review]

**Review by anderseknert — DISMISSED** [review]

**Comment by anderseknert** [comment]
LGTM, I'll let @tsandall merge this since there's an active change request from him, although AFAICS that's been addressed. Thanks for contributing @aflmp !

**Review by tsandall — APPROVED** [review]

**Comment by aflmp** [comment]
rebased and squashed all the commits, ready to be merged. 
thanks everyone

**Comment by anderseknert** [comment]
Thanks for your contribution, Alam! This will be helpful to many 👍

**Pull request merged**